components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/rabbitmq/RabbitMQInjectHandler.java

```diff
@@ -149,58 +88,35 @@ public States invokeAndReturnAckState(RabbitMQMessage message, String inboundNam
                 seq.setErrorHandler(onErrorSeq);
                 msgCtx.setProperty(SynapseConstants.IS_INBOUND, true);
                 msgCtx.setProperty(SynapseConstants.INBOUND_ENDPOINT_NAME, inboundName);
-                msgCtx.setProperty(SynapseConstants.ARTIFACT_NAME, SynapseConstants.FAIL_SAFE_MODE_INBOUND_ENDPOINT + inboundName);
+                msgCtx.setProperty(SynapseConstants.ARTIFACT_NAME,
+                        SynapseConstants.FAIL_SAFE_MODE_INBOUND_ENDPOINT + inboundName);
                 synapseEnvironment.injectInbound(msgCtx, seq, sequential);
             } else {
                 log.error("Sequence: " + injectingSeq + " not found");
             }
 
-            if (readMessageCtxProperty(msgCtx, RabbitMQConstants.SET_ROLLBACK_ONLY, false)) {
-                if (readMessageCtxProperty(msgCtx, RabbitMQConstants.SET_REQUEUE_ON_ROLLBACK, true)) {
-                    return new States(RabbitMQAckStates.REJECT_AND_REQUEUE, false);
-                } else {
-                    return new States(RabbitMQAckStates.REJECT, false);
-                }
-            }
-
-            return new States(RabbitMQAckStates.ACK, false);
-
-        } catch (Exception e) {         //we need to handle any exception upon injecting to mediation
-            if (readMessageCtxProperty(msgCtx, RabbitMQConstants.SET_REQUEUE_ON_ROLLBACK, false)) {
-                log.error("Error while mediating message. Message is requeued as " +
-                        "SET_REQUEUE_ON_ROLLBACK property is set to true.", e);
-                return new States(RabbitMQAckStates.REJECT_AND_REQUEUE, true);
-            }
-            log.error("Error while mediating message. Message is rejected.", e);
-            return new States(RabbitMQAckStates.REJECT, true);
-        }
-    }
-
-    private boolean readMessageCtxProperty(org.apache.synapse.MessageContext msgCtx, String propertyName,
-                                           boolean defaultValue) {
-        Object propertyObj = msgCtx.getProperty(propertyName);
-        if (propertyObj != null) {
-            if ((propertyObj instanceof Boolean && ((Boolean) propertyObj))
-                    || (propertyObj instanceof String && "true".equals(propertyObj))) {
-                return true;
-            }
-            if (propertyObj instanceof Boolean || propertyObj instanceof String && "false".equals(propertyObj)) {
+            Object rollbackProperty = msgCtx.getProperty(RabbitMQConstants.SET_ROLLBACK_ONLY);
+            if ((rollbackProperty instanceof Boolean && ((Boolean) rollbackProperty)) ||
+                    (rollbackProperty instanceof String && Boolean.parseBoolean((String) rollbackProperty))) {
                 return false;
             }
+
```

Comment:
Will this be backward compatible with the removal of the usage of REJECT_AND_REQUEUE?

Reply:
Requeue is handled properly in RabbitMQ with DLX. Let's document it so that existing customers could migrate easily.